lib/group_admin.js

```diff
@@ -38,6 +39,82 @@ GroupAdmin.prototype.describeGroup = function (groupId) {
 };
 
 /**
+ * Provides output similar to
+ * kafka.admin.ConsumerGroupCommand --describe
+ *
+ * @param  {String} groupId
+ * @param  {Array} offsetFetchRequestArgs
+ * @return {Promise}
+ */
+GroupAdmin.prototype.fetchConsumerLag = function (groupId, offsetFetchRequestArgs) {
+    return Promise.all([
+        this.client.offsetFetchRequestV1(groupId, offsetFetchRequestArgs),
+        this._fetchHighWaterMark(offsetFetchRequestArgs)
+    ]).spread(function (offsets, highWaterMark) {
+        var i, j, res;
+        var responses = [];
+
+        // Match the returned data into a coherent response
+        for (i = 0; i < offsets.length; i++) {
+            res = {
+                topic: offsets[i].topic,
+                partition: offsets[i].partition,
+                offset: offsets[i].offset,
+                highwaterMark: null,
+                consumerLag: null
+            };
+
+            for (j = 0; j < highWaterMark.length; j++) {
+                if (
+                    offsets[i].topic === highWaterMark[j].topic
+                    && offsets[i].partition === highWaterMark[j].partition
+                ) {
+                    res.highwaterMark = highWaterMark[j].highWaterMark;
+                    if (res.offset > 0 && res.highwaterMark > 0) {
+                        res.consumerLag = res.highwaterMark - res.offset;
+                    }
+                }
+            }
+
+            responses.push(res);
+        }
+
+        return responses;
+    });
+};
+
+GroupAdmin.prototype._fetchHighWaterMark = function (offsetFetchRequestArgs) {
+    var topicName, topicPartitions, i, j;
+    var offsetRequestsPromises = [];
+
+    for (i = 0; i < offsetFetchRequestArgs.length; i++) {
+        topicName = offsetFetchRequestArgs[i].topicName;
+        topicPartitions = offsetFetchRequestArgs[i].partitions;
+
+        for (j = 0; j < topicPartitions.length; j++) {
+            offsetRequestsPromises.push(
+                this._fetchHighWaterMarkOffset(topicName, topicPartitions[j])
+            );
+        }
+    }
+
+    return Promise.all(offsetRequestsPromises);
```

Comment:
Wouldn't the kafka client queue up the requests internally? And what about clusters with many broker nodes, in this case firing many requests simultaneously might be beneficial. And exactly how much parallelism is needed? Seem we _might_ be going for premature optimization, without first knowing the answer to the latter question.

I have no kafka operating experience yet, but I will soon be testing this in replicated production environment. So if there're any improvement to be made, I'll gladly share them. I will only have 12 partitions and around 10 topics, might as well leave this for guys running bigger clusters to figure out.